_posts/2020-06-18-a-sustainable-nuget-marketplace-will-have-to-compete-with-the-nuget-gallery.markdown

```diff
@@ -0,0 +1,368 @@
+
+I know when I was building [MyGet](https://www.myget.org/) with [Xavier](https://twitter.com/xavierdecoster),
+we dreamt up something like a NuGet marketplace as well. Will have to check but I think I still have `nugetmarketplace.com`
+somewhere...
+
+The above is scientific proof a NuGet marketplace is needed!
+Let's look at what a NuGet marketplace should help with.
+
+### For producers: legal services
+
+Open source projects can still be open source, but sell additional services. These could be:
+
+* Dual-licensed open source
+* Paid add-on packages
+* Support services
+* Consulting services
+
+Dual-licensing will require a set of templates that projects can work with.
+If the project uses Apache-2, MIT, GPL, or other open source licenses, what will a commercial dual-license
+look like?
+
+Paid add-on packages may be open or closed source, but they will still need a license. A NuGet marketplace
+will require templates and guidance.
+
+Support services and consulting services are harder. How will a NuGet marketplace assist with these?
+Contract templates and guidance, again, may be of help here.
+
+The legal aspect will be a bit like what the open source foundations offer:
+help with protecting producers and consumers.
+
+There is one additional aspect a NuGet marketplace will have to help with: establishing rules and preventing
+their abuse. What if I have an MIT-licensed project, and someone sells it with a dual license on this
+marketplace?
+
+<blockquote class="twitter-tweet"><p lang="en" dir="ltr">It will be perfectly legal and valid for someone to sell your MIT licence code. You expressly permitted that.</p>&mdash; Damian Hickey (@randompunter) <a href="https://twitter.com/randompunter/status/1273173878752370688?ref_src=twsrc%5Etfw">June 17, 2020</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+Morally, this is "not done", but legally, this may happen. The marketplace will have to prevent this and have
+rules and guidance on how to go about this. We want sustainability!
+
+Another similarity with open source foundations is in the legality of purchasing packages.
+A NuGet marketplace will have to make sure consumers who purchase here can trust their purchase is legal
+and they can use the package without issue, now and in the future.
+
+Others are GDPR (as a consumer, is my personal data safe?), and probably some more. Feel free to add them
+in the comments and I will update the post.
+
+### For producers: business services
+
+Before we can sell packages, what's the legal status of an open source project? Is it a company? A non-profit?
+
+A NuGet marketplace will probably be the legal entity where consumers purchase packages from.
+Much like how [Fastspring](https://www.fastspring.com/) acts as a reseller, a NuGet marketplace will have to
+do this as well.
+
+A NuGet marketplace will have to act as a reseller, for several reasons:
```

Comment:
"Merchant of Record" is the legal term - the Apple app store and others also have to do this. Without doing this, the legal / compliance costs for supporting each vendor explode out of control.